draft-xyz-rats-psa-endorsements.md

```diff
@@ -131,94 +141,173 @@ firmware measurement associated with Implementation ID
+{{!RFC8152}} and wrapped in the `COSE_KeySet`. The number of items in the
+`COSE_KeySet` MUST be 1.
+
+The example in {{ex-identity-claim}} shows a CoMID identity claim carrying a
+secp256r1 EC public IAK associated with Instance ID `4ca3...d296`.
+
+An Identity CoMID can carry as many Identity Claims as needed.
 
-~~~ goat
-{::include art/updates-patches.txt}
 ~~~
-{: #fig-updates-patches title="Updates, Patches Relations and Semantic Versioning"}
+{::include examples/instance-pub.diag}
+~~~
+{: #ex-identity-claim title="Example Identity Claim"}
 
-The Reference Value CoMID of the patching or updating firmware has a
-`linked-tag-entry` map populated as follows:
 
-* key 0 contains the tag identifier of the Reference Value CoMID of the patched
-  or updated firmware;
-* key 1 contains one of `comid-patches` or `comid-updates` relations.
+## Certification
+{: #sec-certificates}
 
-{{ex-linked-tag}} provides an example of a Reference Value CoMID patching
-another Reference Value CoMID with a UUID tag identifier
-`3f06af63-a93c-11e4-9797-00505690773f`.
+A PSA Certification can certify PSA hardware, software RoT or the complete device.
+Specifically, it can certify one of the following:
+
+* The underlying platform, identified by the Impementation ID
+
+* One or more firmware components of the mutable PSA RoT.
+
+* To entire PSA device.
+
+PSA Certification claim contains the metadata about the certificate. The claim is 
+expressed by creating a new CoMID Tag for provisioning certification details.
+
+The certification status is encoded as a CoMID endorsement as follows:
+
+* The Implementation ID of the PSA RoT to which the certification metadata apply, is encoded 
+  using the class-id-type-choice (2) entry in the `element-name` map with a `tagged-impl-id` type;
+
+* The metadata associated with the certification is encoded in a `comid.psa-cert-meta` structure
+  which extends the `endorsed-value-map`.
+  
+* Optionally, the CoMID linked tags can contain one or more entries of linked-tag-map, each entry carries
+  tag id of CoMID, which it certifies and tag-rel type set to `comid.supplements` relation,
+  as illustrated in {{fig-cert-link}}.
+
+A PSA Certification data element has the following attributes:
+
+* `cert-level-type-choice`    : PSA certification level.  Acceptable values are x, y, z...
+* `cert-num-type`             : A unique number for each certificate
+* `cert-issue-date-type`      : Certificate date of issue, formatted as per RFC 3339 
+* `cert-test-lab-type`        : Name of the certication lab
+* `cert-holder-type`          : Name of Organization which holds the certificate
+* `cert-product-type`         : Name of the product been certified
+* `cert-hw-version-type`      : Version of Hardware certified
+* `cert-sw-version-type`      : Version of Software certified
+* `cert-type-type`            : Further details on certification
+* `cert-dev-type-type-choice` : Type of certificate, Chip, Device or System Software
```

Comment:
```suggestion
* `$cert-dev-type-type-choice` : Type of certificate.  Acceptable values are: Chip, Device or System Software
```

Reply:
Again I would prefer not to use $ in the explanation!

Reply:
ditto

Reply:
I would prefer to remove $ reference from all CDDL names here unilaterally and leave it to CDDL sections, just makes reading simpler and clear for the reader!

Reply:
see https://github.com/thomas-fossati/draft-psa-endorsements/pull/14#discussion_r621160154